source/common/grpc/grpc_web_filter.cc

```diff
@@ -87,6 +90,13 @@ Http::FilterDataStatus GrpcWebFilter::decodeData(Buffer::Instance& data, bool) {
   const std::string decoded = Base64::decode(
       std::string(static_cast<const char*>(decoding_buffer_.linearize(decoding_buffer_.length())),
                   decoding_buffer_.length()));
+  if (decoded == EMPTY_STRING) {
+    // Error happened when decoding base64.
+    Http::Utility::sendLocalReply(*decoder_callbacks_, stream_destroyed_, Http::Code::BadRequest,
```

Comment:
Are the error codes for gRPC-Web formalized anywhere? Or is it "just do what makes sense according to HTTP, and clients should probably also do what makes sense according to HTTP, and it'll Just Work"?